### PR TITLE
Pillar of the Paruns custom mode

### PR DESCRIPTION
**Pull request by Susucre**
~~This requires #10629 and #10632 first. I'll rebase once a version of those get merged.~~

So I am working on a custom game mode. Is it alright to merge it in the code base, or should I convince some server maintainers to add the .jar to their server's plugins?
Once we have [custom options for the number of starting cards in hand](https://github.com/magefree/mage/issues/10551) and [Emblem cards](#10498), I could replace the custom mode by a custom emblem to despecialize the game mode.

As for the custom game mode, here are the changes from an usual limited match:
- The starting hand size is 6
- At the beginning of each player's first main step, that player conjure Pillar of the Paruns on the battlefield. This counts as a land played for the turn.

  - [x] Prefix the mode with Custom
  - [x] Add Comment to describe the game mode in the code
  - [x] Test Karn Liberated restarting the game.

### CODE REVIEW

**Comment by Susucre** [comment]
[[Pillar of the Paruns]]

File: Mage.Server.Plugins/Mage.Game.PillarOfTheParunsDuel/src/mage/game/PillarOfTheParunsDuel.java [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+import mage.game.mulligan.Mulligan;
+import mage.game.turn.TurnMod;
+import mage.players.Player;
+
+import java.util.UUID;
+
+/**
+ * @author Susucr
+ */
+public class PillarOfTheParunsDuel extends GameImpl {
```
Comment (JayDi85) [inline Mage.Server.Plugins/Mage.Game.PillarOfTheParunsDuel/src/mage/game/PillarOfTheParunsDuel.java:19]: Must add game mode description, rules and other information here as docs/comments. If it's an official/public then must contains link to the public page/wiki.

**Review by JayDi85 — CHANGES_REQUESTED** [review]

**Comment by JayDi85** [comment]
Also please test that game mode and effects with [[Karn Liberated]] -- is it works fine and without errors after game restart.

File: .gitignore [hunk 2]
```diff
@@ -53,6 +53,7 @@ Mage.Server.Plugins/Mage.Game.PennyDreadfulCommanderFreeForAll/target
 Mage.Server.Plugins/Mage.Game.FreeformCommanderDuel/target/
 Mage.Server.Plugins/Mage.Game.FreeformCommanderFreeForAll/target/
 Mage.Server.Plugins/Mage.Game.FreeformUnlimitedCommander/target/
+Mage.Server.Plugins/Mage.Game.PillarOfTheParunsDuel/target/
```
[the remark this answers is not in the document]
Comment (Susucre) [reply .gitignore:56]: ok! This is definitively a custom format and not an official one.

**Comment by Susucre** [comment]
Oh right, it is possible to restart the game that way, will definitively test that.

**Review by JayDi85 — APPROVED** [review]

**Pull request merged**